src/fortran/fdb.f90

```diff
@@ -1,39 +1,329 @@
+         integer(kind=c_long), intent(in), value  :: count
+         integer(kind=c_long), intent(inout)      :: read
+      end function fdb_datareader_read
+   end interface
+
+   interface
+      integer(kind=c_int) function fdb_delete_datareader(dr) bind(C,name='fdb_delete_datareader')
+         use, intrinsic :: iso_c_binding, only : c_int, c_ptr
+         type(c_ptr), intent(in), value      :: dr
+      end function fdb_delete_datareader
+   end interface
+
+
+   interface
+      integer(kind=c_int) function fdb_list(fdb_handle, req, it, duplicates) bind(C,name='fdb_list')
+         use, intrinsic :: iso_c_binding, only : c_int, c_ptr, c_bool, c_int
+         type(c_ptr), value   :: fdb_handle
+         type(c_ptr), value   :: req 
+         type(c_ptr)       :: it
+         logical(kind=c_bool) :: duplicates
+      end function fdb_list
+   end interface
+
+   interface
+      integer(kind=c_int) function fdb_new_listiterator(it) bind(C,name='fdb_new_listiterator')
+         use, intrinsic :: iso_c_binding, only : c_int, c_ptr
+         type(c_ptr), intent(in)      :: it
+      end function fdb_new_listiterator
+   end interface
+
+   interface
+      integer(kind=c_int) function fdb_listiterator_next(it) bind(C,name='fdb_listiterator_next')
+         use, intrinsic :: iso_c_binding, only : c_int, c_ptr, c_char, c_bool
+         type(c_ptr), intent(in), value               :: it
+      end function fdb_listiterator_next
+   end interface
+
+   interface
+      integer(kind=c_int) function fdb_delete_listiterator(it) bind(C,name='fdb_delete_listiterator')
+         use, intrinsic :: iso_c_binding, only : c_int, c_ptr
+         type(c_ptr), intent(in), value      :: it
+      end function fdb_delete_listiterator
+   end interface
+
+   interface
+      integer(kind=c_int) function fdb_listiterator_attrs(it, uri, off, len) bind(C,name='fdb_listiterator_attrs')
+         use, intrinsic :: iso_c_binding, only : c_int, c_ptr, c_int, c_char
+         type(c_ptr), intent(in), value      :: it
+         character(kind=c_char, len=1), dimension(*), INTENT(inout) :: uri
+         integer(kind=c_int), intent(inout) :: off
+         integer(kind=c_int), intent(inout) :: len
+      end function fdb_listiterator_attrs
+   end interface
+
+   CONTAINS
+   SUBROUTINE get_value_of_key(igrib, keyname, keyvalue_str, keyvalue_int)
+      character(len=*), INTENT(IN)         :: keyname
+      character(len=128), INTENT(OUT)    :: keyvalue_str
+      integer, INTENT(INOUT), OPTIONAL     :: keyvalue_int
+      integer, INTENT(IN)                 :: igrib
+      integer                             :: res
+      integer                             :: is_missing
+      call codes_is_missing(igrib, trim(keyname), is_missing);
+      if (is_missing /= 1) then
+         ! key value is not missing so get values
+         if (present(keyvalue_int)) then
+            call codes_get_int(igrib, keyname, keyvalue_int)
```

Comment:
what happens if you try to call codes_get_int on a key that is can not be casted into int ? I guess you get an error code? so probably we should check for error codes ?

Reply:
I just tried with typeOfFirstFixedSurface, which normally returns a string. It doesnt raise an error code, it just returns an integer instead of a string. So i guess the error codes in status from codes_get_int are for other cases(?)

Reply:
I think there is no point in checking the status since eccodes will write an error message anyway f the status is non zero? Or should we exit() if a status is returned?